com.unity.render-pipelines.core/Runtime/Common/CoreAttributes.cs

```diff
@@ -2,6 +2,36 @@
 
 namespace UnityEngine.Rendering
 {
+    /// <summary>
+    /// This attribute allows you to add information for a class to be supported on a render pipeline
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class SupportedOnAttribute : Attribute
```

Comment:
This remind me that there is this attribute used sparcely that rely on the asset. Perhaps we should rework it to use your introduced attribute instead ?
> [ScriptableRenderPipelineExtension(typeof(HDRenderPipelineAsset))]

It is in trunk. It is based on the type of asset in Graphics Settings. I had totally forgotten I done this some years ago :(
@jenniferd-unity for awareness